### PR TITLE
Fixes #6786

### LINKED ISSUE #6786: 'Session closed. Most likely the page has been closed' error seen when browser.mock called after browser.sessionReload
**Environment:**
 - **WebdriverIO version:** v7 (but seen on v6 as well)
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 12.18.3
 - **NPM version:** 6.14.11
 - **Browser name and version:** Chrome 90
 - **Platform name and version:** Windows 10
 - **Additional wdio packages used (if applicable):** 
    - @wdio/cucumber-framework
    - @wdio/local-runner
    - @wdio/selenium-standalone-service
    - @wdio/sync 

**Config of WebdriverIO**: Its nothing bespoke, we run a cucumber based framework and since its a E2E framework we have a few applications on different domain and reloading session (closing & opening browser) helps us reset the state better.
```javascript
    afterScenario: function (world) {
        browser.reloadSession();
    },
```

**Describe the bug**
When using `browser.mock()` after calling `browser.sessionReload()`, I am seeing the below error message:

```javascript
Error: Protocol error (Runtime.callFunctionOn): Session closed. Most likely the page has been closed.
    at CDPSession.send (C:\Users\user\workspace\node.js\browser.mock-defect\node_modules\puppeteer-core\src\common\Connection.ts:251:9)
    at ExecutionContext._evaluateInternal (C:\Users\user\workspace\node.js\browser.mock-defect\node_modules\puppeteer-core\src\common\ExecutionContext.ts:252:44)
    at ExecutionContext.evaluate (C:\Users\user\workspace\node.js\browser.mock-defect\node_modules\puppeteer-core\src\common\ExecutionContext.ts:138:23)
    at DOMWorld.evaluate (C:\Users\user\workspace\node.js\browser.mock-defect\node_modules\puppeteer-core\src\common\DOMWorld.ts:163:20)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at Browser.runCommandWithHooks (C:\Users\user\workspace\node.js\browser.mock-defect\node_modules\@wdio\sync\build\wrapCommand.js:100:24)
    at Browser.wrapCommandFn (C:\Users\user\workspace\node.js\browser.mock-defect\node_modules\@wdio\sync\build\wrapCommand.js:67:44)
    at World.<anonymous> (C:\Users\user\workspace\node.js\browser.mock-defect\features\step-definitions\steps.js:11:29)
    at World.executeSync (C:\Users\user\workspace\node.js\browser.mock-defect\node_modules\@wdio\sync\build\index.js:38:22)
    at C:\Users\user\workspace\node.js\browser.mock-defect\node_modules\@wdio\sync\build\index.js:69:68
```

**To Reproduce**
Steps to reproduce the behaviour:
1. Setup two scenarios that use `browser.mock()` to spy on a network request
2. Now in the wdio.conf.js file, in `afterScenario()` hook call `browser.sessionReload()`
3. When you run these tests, the first test will pass but the 2nd will fail with the error seen above.

**Expected behaviour**
It should work like the first test.

**Log**
https://gist.github.com/johnson-tyl/e50752561d577f0e88c11f11528eefed

I have attached a sample project where you can see this happening:
[browser.mock-defect.zip](https://github.com/webdriverio/webdriverio/files/6406927/browser.mock-defect.zip)
You can run `npm i` followed by `npm test` to run the test zip file.

**NOTE**: The first test also fails in the sample project but the reason for that is different, there not much to spy on on this page, so probably it goes too quickly for mock to detect its already come back with 200.

### PR DESCRIPTION
**Pull request by jonn-set**
## Proposed changes

Fixes #6786 

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

Its a simple one line fix on session reload,

### CODE REVIEW

File: packages/webdriverio/src/commands/browser/reloadSession.ts [hunk 1]
```diff
@@ -41,6 +41,9 @@ export default async function reloadSession (this: WebdriverIO.Browser) {
         log.warn(`Suppressing error closing the session: ${err.stack}`)
     }
 
+    if (this.puppeteer)
+        this.puppeteer = undefined;
```
Comment (christian-bromann) [inline packages/webdriverio/src/commands/browser/reloadSession.ts:45]:
```suggestion
    if (this.puppeteer) {
        delete this.puppeteer
    }
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Can we have a test for this? I would advocate for a unit test.

File: packages/webdriverio/src/commands/browser/reloadSession.ts (hunk 1 above)
Comment (jonn-set) [reply packages/webdriverio/src/commands/browser/reloadSession.ts:45]: That does not work for some reason. I get this error on the 2nd `browser.url` call, first one goes through without any issues. 

image

This is a sample project I am running by linking the package.

**Stacktrace**:
```javascript
[0-0] Error in "1: I search for 'reloadSession'"
invalid session id: No active session with ID 78bdf25b0b241b231f6b3fec05a7b5b5
    at Object.getErrorFromResponseBody (C:\Users\jonne\workspace\node.js\wdio-jonn\packages\webdriver\src\utils.ts:221:12)
    at WebDriverRequest._request (C:\Users\jonne\workspace\node.js\wdio-jonn\packages\webdriver\src\request.ts:195:23)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at Browser.runCommandWithHooks (C:\Users\jonne\workspace\node.js\wdio-jonn\packages\wdio-sync\src\wrapCommand.ts:125:25)
    at Browser.runCommandWithHooks (C:\Users\jonne\workspace\node.js\wdio-jonn\packages\wdio-sync\src\wrapCommand.ts:118:24)
    at Browser.wrapCommandFn (C:\Users\jonne\workspace\node.js\wdio-jonn\packages\wdio-sync\src\wrapCommand.ts:78:44)
    at World.<anonymous> (C:\Users\jonne\workspace\node.js\browser.mock-defect\features\step-definitions\steps.js:7:13)
    at World.executeSync (C:\Users\jonne\workspace\node.js\browser.mock-defect\node_modules\@wdio\sync\build\index.js:38:22)
    at C:\Users\jonne\workspace\node.js\browser.mock-defect\node_modules\@wdio\sync\build\index.js:69:68

```

Comment (jonn-set) [reply packages/webdriverio/src/commands/browser/reloadSession.ts:45]: @christian-bromann What next, do you want to try something else?

**Comment by jonn-set** [comment]
@christian-bromann I have changed it to check for puppeteer to be connected instead of the variable to exist. Check if you are happy with it please?

**Comment by jonn-set** [comment]
@christian-bromann added unit tests to increase coverage, also fixed the broken mocks, let me know if you happy with the overall solution & test coverage please?

**Comment by jonn-set** [comment]
@christian-bromann bump

**Review by christian-bromann — APPROVED** [review]
👍  Thanks a lot!

**Comment by jonn-set** [comment]
@christian-bromann Thanks for merging. Can I ask when's the next release, please?

**Comment by christian-bromann** [comment]
It was suppose to happen already but the [pipeline failed](https://github.com/webdriverio/webdriverio/runs/2663120095?check_suite_focus=true). Looking into it.

**Pull request merged**